packages/compiler/src/experimental/typekit/kits/type.ts

```diff
@@ -272,5 +274,39 @@ defineKit<TypekitExtension>({
     isUserDefined(type) {
       return getLocationContext(this.program, type).type === "project";
     },
+    inNamespace(type: Type, namespace: Namespace): boolean {
+      if (type === namespace) {
+        return true;
+      }
+
+      // Handle types that can be sourced from other types
+      if (type.kind === "ModelProperty") {
```

Comment:
Definitely going to split up this ternary - I know it's hard to read (at least for me!)

Reply:
I would also change this to just a swtich statement on the `type.kind`

Reply:
I was _literally_ just doing that 🤝 😄